src/test/java/org/openstreetmap/atlas/geography/atlas/command/AtlasFeatureCountsSubCommandTestCase.java

```diff
@@ -0,0 +1,74 @@
+package org.openstreetmap.atlas.geography.atlas.command;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.openstreetmap.atlas.streaming.resource.File;
+
+/**
+ * Test case verifying that protecting against multiple concurrent threads in the AtlasReader report
+ * the right output AtlasFeatureCountsSubCommand class
+ * 
+ * @author cstaylor
+ */
+public class AtlasFeatureCountsSubCommandTestCase
+{
+    @ClassRule
+    public static AtlasFeatureCountsSubCommandTestCaseRule setup = new AtlasFeatureCountsSubCommandTestCaseRule();
+
+    private static Path temporaryPath;
+
+    @AfterClass
+    public static void cleanupAtlasesOnDisk() throws IOException
+    {
+        Files.walk(temporaryPath).filter(Files::isRegularFile).forEach(path ->
+        {
+            try
+            {
+                Files.deleteIfExists(path);
+            }
+            catch (final IOException oops)
+            {
+                throw new RuntimeException("Error when cleaning up", oops);
+            }
+        });
+        Files.deleteIfExists(temporaryPath);
+    }
+
+    @BeforeClass
+    public static void prepareAtlasesOnDisk() throws IOException
+    {
+        final Path path = Files.createTempDirectory("atlastest");
```

Comment:
Can we use the File.temporaryFolder() here, so we are using our internal File code without switching between ours and nio?